CRM/Civicaseextras/AngularModifiers/CaseDuration.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+use Civi\Angular\Manager as AngularManager;
+use Civi\Angular\ChangeSet as AngularChangeSet;
+use CRM_Civicaseextras_Services_CustomValueService as CustomValueService;
+
+class CRM_Civicaseextras_AngularModifiers_CaseDuration {
+
+  /**
+   * @var AngularManager
+   *   A reference to the Angular Manager object as provided by the alter angular hook.
+   */
+  protected $angular;
+
+  /**
+   * @var array
+   *   The case duration custom field information.
+   */
+  protected $caseDuration;
+
+  /**
+   * @param AngularManager $angular as provided by the alter angular hook.
+   * @param CustomValueService $customValueService an instance of the custom value service.
+   */
+  public function __construct(AngularManager &$angular, CustomValueService $customValueService) {
+    $this->angular = $angular;
+    $this->caseDuration = $customValueService->getCustomField('case_stats', 'duration');
```

Comment:
Mmm, I really like that you're using dependency injection now, but how about instead of fetching stuff at instantiation you fetch it later when it's needed? You could fetch the `caseDuration` in `runModifications` and just pass it into the other two methods.